core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -1136,7 +1145,45 @@ wasmtime_ssp_fd_renumber(
     fd_object_release(fo);
     --ft->used;
 
+    // Handle renumbering of any preopened resources
```

Comment:
Added logic for renumbering preopened resources so this [test](https://github.com/WebAssembly/wasi-testsuite/blob/main/tests/rust/src/bin/overwrite_preopen.rs) passes. I don't think there's any other reason not to support renumbering preopened resources.